actions/service.go

```diff
@@ -174,66 +183,101 @@ func (s *Service) runTasks(ctx context.Context, record graveler.HookRecord, task
-			for _, task := range actionTasks {
-				if task.StartTime.IsZero() {
-					break
-				}
-				taskPassed := task.Err == nil
-				_, err = tx.Exec(`INSERT INTO actions_run_hooks(repository_id, run_id, hook_run_id, action_name, hook_id, start_time, end_time, passed)
-			VALUES ($1,$2,$3,$4,$5,$6,$7,$8)`,
-					record.RepositoryID, task.RunID, task.HookRunID, task.Action.Name, task.HookID, task.StartTime, task.EndTime, taskPassed)
-				if err != nil {
-					return nil, fmt.Errorf("insert run hook information (%s %s): %w", task.Action.Name, task.HookID, err)
-				}
+		for _, hookRun := range manifest.HooksRun {
+			_, err = tx.Exec(`INSERT INTO actions_run_hooks(repository_id, run_id, hook_run_id, action_name, hook_id, start_time, end_time, passed)
+				VALUES ($1,$2,$3,$4,$5,$6,$7,$8)`,
+				repositoryID, hookRun.RunID, hookRun.HookRunID, hookRun.ActionName, hookRun.HookID, hookRun.StartTime, hookRun.EndTime, hookRun.Passed)
+			if err != nil {
+				return nil, fmt.Errorf("insert run hook information %s/%s: %w", hookRun.RunID, hookRun.HookRunID, err)
 			}
 		}
 		return nil, nil
 	}, db.WithContext(ctx))
 	return err
 }
 
-func runInformationBasedOnTasks(tasks [][]*Task) (startTime time.Time, endTime time.Time, passed bool) {
-	passed = true
+func buildRunManifest(record graveler.HookRecord, tasks [][]*Task) runManifest {
+	manifest := runManifest{
+		Run: RunResult{
+			RunID:     record.RunID,
+			BranchID:  record.BranchID.String(),
+			SourceRef: record.SourceRef.String(),
+			EventType: string(record.EventType),
+			Passed:    true,
+			CommitID:  record.CommitID.String(),
+		},
+	}
 	for _, actionTasks := range tasks {
 		for _, task := range actionTasks {
 			// skip scan when task didn't run
 			if task.StartTime.IsZero() {
 				break
 			}
-			// keep min start time
-			if startTime.IsZero() || task.StartTime.Before(startTime) {
-				startTime = task.StartTime
+			// record hook run information
+			manifest.HooksRun = append(manifest.HooksRun, TaskResult{
+				RunID:      task.RunID,
+				HookRunID:  task.HookRunID,
+				HookID:     task.HookID,
+				ActionName: task.Action.Name,
+				StartTime:  task.StartTime,
+				EndTime:    task.EndTime,
+				Passed:     task.Err == nil,
+			})
+			// keep min run start time
+			if manifest.Run.StartTime.IsZero() || task.StartTime.Before(manifest.Run.StartTime) {
+				manifest.Run.StartTime = task.StartTime
 			}
-			// keep max end time
-			if endTime.IsZero() || task.EndTime.After(endTime) {
-				endTime = task.EndTime
+			// keep max run end time
+			if manifest.Run.EndTime.IsZero() || task.EndTime.After(manifest.Run.EndTime) {
+				manifest.Run.EndTime = task.EndTime
 			}
 			// did we failed
-			if passed && task.Err != nil {
-				passed = false
+			if manifest.Run.Passed && task.Err != nil {
+				manifest.Run.Passed = false
 			}
 		}
 	}
-	return
+
+	return manifest
 }
 
 func (s *Service) UpdateCommitID(ctx context.Context, repositoryID string, runID string, commitID string) error {
```

Comment:
This needs to update OS too.

Reply:
I'll read the file and update the commit ID - not sure I like to re-write but it will reflect the same information as you said.

Reply:
Updated